File: fixtures/sample.dart
// Sample Dart file for fmm parser validation
// Covers: classes, abstract classes, mixins, enums, extensions,
// typedefs, functions, variables, imports, privacy convention

import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:async';
import 'dart:convert';
import './relative_file.dart';
import '../utils/helpers.dart';

// Public classes

class NetworkManager {
  Future<void> fetchData() async {}

  void _privateMethod() {}

  void cancelAll() {}
}

abstract class BaseWidget extends StatelessWidget {
  Widget build(BuildContext context);
}

class UserProfile {
  final String name;
  final String email;

  UserProfile({required this.name, required this.email});
}

// Private class (underscore prefix)
class _InternalHelper {
  void doWork() {}
}

// Mixin

mixin Loggable {
  void log(String message) {
    print(message);
  }
}

mixin Cacheable on Object {
  void invalidateCache() {}
}

// Enum

enum Direction { north, south, east, west }

enum HttpStatus {
  ok(200),
  notFound(404),
  serverError(500);

  final int code;
  const HttpStatus(this.code);
}

// Extension

extension StringExtension on String {
  String capitalize() => '${this[0].toUpperCase()}${substring(1)}';

  bool get isBlank => trim().isEmpty;
}

extension IntExtension on int {
  bool get isEven2 => this % 2 == 0;
}

// Typedef

typedef Callback = void Function(String);
typedef JsonMap = Map<String, dynamic>;

typedef _PrivateCallback = void Function(int);

// Top-level functions

void globalFunction() {}

String processData(List<String> items) {
  return items.join(', ');
}

void _privateFunction() {}

Future<void> asyncOperation() async {}

// Top-level variables

final String appVersion = '1.0.0';
const int maxRetries = 3;
var isDebugMode = false;

final _privateVar = 'hidden';

File: fixtures/sample.kt
// Sample Kotlin file for fmm parser validation
// Covers: classes, interfaces, objects, data classes, sealed classes,
// enum classes, functions, properties, typealiases, imports, visibility

package com.example.app

import kotlin.collections.List
import kotlin.collections.Map
import java.util.UUID
import java.util.concurrent.Executor
import org.example.utils.Logger

// Public classes (default visibility = public)

class NetworkManager {
    fun fetchData(): ByteArray? = null

    private fun internalRetry() {}

    fun cancelAll() {}
}

open class BaseRepository {
    open fun findById(id: String): Any? = null
}

// Data classes

data class UserProfile(
    val name: String,
    val email: String,
    val age: Int
)

data class APIResponse<T>(
    val data: T?,
    val error: String?
)

// Sealed class

sealed class Result<out T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Failure(val error: Throwable) : Result<Nothing>()
    object Loading : Result<Nothing>()
}

// Interfaces

interface Repository<T> {
    fun findAll(): List<T>
    fun save(item: T): Boolean
}

interface Cacheable {
    fun invalidateCache()
}

// Objects

object AppConfig {
    val baseURL = "https://api.example.com"
    val timeout = 30_000L
}

object DatabaseManager {
    fun getConnection(): Any = TODO()
}

// Enum class

enum class Direction {
    NORTH, SOUTH, EAST, WEST
}

enum class HttpStatus(val code: Int) {
    OK(200),
    NOT_FOUND(404),
    SERVER_ERROR(500)
}

// Private and internal (should NOT be exported)

private class InternalHelper {
    fun process() {}
}

internal class ModuleInternal {
    fun doWork() {}
}

private fun hiddenFunction(): String = "hidden"
internal fun moduleFunction(): Int = 42

// Top-level public declarations

fun createManager(config: Map<String, Any>): NetworkManager = NetworkManager()

fun processData(input: List<String>): List<String> = input.map { it.uppercase() }

val MAX_RETRIES = 3
var isDebugMode = false
val VERSION = "1.0.0"

typealias StringMap = Map<String, String>
typealias Callback<T> = (T) -> Unit

// Class with companion object

class ServiceLocator {
    companion object {
        const val TAG = "ServiceLocator"
        fun getInstance(): ServiceLocator = ServiceLocator()
    }

    fun resolve(name: String): Any? = null
}

// Annotated declarations

@Deprecated("Use newMethod instead")
fun oldMethod(): Unit {}

suspend fun asyncOperation(): String = "result"
